test/new-e2e/pkg/environments/aws/kubernetes/kindvm.go

```diff
@@ -0,0 +1,199 @@
+	vmOptions         []ec2.VMOption
+	agentOptions      []kubernetesagentparams.Option
+	fakeintakeOptions []fakeintake.Option
+	extraConfigParams runner.ConfigMap
+}
+
+func newProvisionerParams() *ProvisionerParams {
+	return &ProvisionerParams{
+		name:              defaultVMName,
+		vmOptions:         []ec2.VMOption{},
+		agentOptions:      []kubernetesagentparams.Option{},
+		fakeintakeOptions: []fakeintake.Option{},
+		extraConfigParams: runner.ConfigMap{},
+	}
+}
+
+// ProvisionerOption is a function that modifies the ProvisionerParams
+type ProvisionerOption func(*ProvisionerParams) error
+
+// WithName sets the name of the provisioner
+func WithName(name string) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.name = name
+		return nil
+	}
+}
+
+// WithEC2VMOptions adds options to the EC2 VM
+func WithEC2VMOptions(opts ...ec2.VMOption) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.vmOptions = opts
+		return nil
+	}
+}
+
+// WithAgentOptions adds options to the agent
+func WithAgentOptions(opts ...kubernetesagentparams.Option) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = opts
+		return nil
+	}
+}
+
+// WithFakeIntakeOptions adds options to the fake intake
+func WithFakeIntakeOptions(opts ...fakeintake.Option) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = opts
+		return nil
+	}
+}
+
+// WithoutFakeIntake removes the fake intake
+func WithoutFakeIntake() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = nil
+		return nil
+	}
+}
+
+// WithoutAgent removes the agent
+func WithoutAgent() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = nil
+		return nil
+	}
+}
+
+// WithExtraConfigParams adds extra config parameters to the environment
+func WithExtraConfigParams(configMap runner.ConfigMap) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.extraConfigParams = configMap
+		return nil
+	}
+}
+
+// Provisioner creates a new provisioner
+func Provisioner(opts ...ProvisionerOption) e2e.TypedProvisioner[environments.KindVMHost] {
+	params := newProvisionerParams()
+	_ = optional.ApplyOptions(params, opts)
+
+	provisioner := e2e.NewTypedPulumiProvisioner(provisionerBaseID+params.name, func(ctx *pulumi.Context, env *environments.KindVMHost) error {
+		params := newProvisionerParams()
+		_ = optional.ApplyOptions(params, opts)
+
+		return KindVMRunFunc(ctx, env, params)
+	}, params.extraConfigParams)
+
+	return provisioner
+}
+
+// KindVMRunFunc is the Pulumi run function that runs the provisioner
+func KindVMRunFunc(ctx *pulumi.Context, env *environments.KindVMHost, params *ProvisionerParams) error {
+
+	awsEnv, err := aws.NewEnvironment(ctx)
+	if err != nil {
+		return err
+	}
+
+	host, err := ec2.NewVM(awsEnv, params.name, params.vmOptions...)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
We should default to `AmazonECS`, don't know if it's already the case

Reply:
I will do it yep!

Reply:
Actually we do not have a way yet to detect if docker is already installed or not and [NewKindCluster](https://github.com/DataDog/test-infra-definitions/blob/main/components/kubernetes/kind.go#L29) func install it by default. Unless we have a way to not install docker when it is already installed we cannot use `AmazonECS`. Created this card to try to solve this: https://datadoghq.atlassian.net/jira/software/c/projects/APL/boards/5339?selectedIssue=APL-2873
But I think this is not straighforward. A not so hard workaround would be to change `NewKindCluster` function to take the `installDocker` args to determine if we install docker or not